nss/integration-tests/integration_test.go

```diff
@@ -153,23 +156,30 @@ func TestIntegration(t *testing.T) {
 }
 
 func TestMain(m *testing.M) {
-	// Build the NSS library and executable in a temporary directory and allow linking to it.
-	tmpDir, cleanup, err := createTempDir()
-	if err != nil {
-		os.Exit(1)
-	}
-	defer cleanup()
-
-	targetDir, libPath = tmpDir, filepath.Join(tmpDir, "libnss_aad.so.2")
-	// Builds the NSS Library.
-	if err = buildRustNSSLib(); err != nil {
-		cleanup()
-		fmt.Fprintf(os.Stderr, "Can not build rust nss library: %v", err)
-		os.Exit(1)
-	}
-
 	testutils.InstallUpdateFlag()
 	flag.Parse()
 
-	m.Run()
+	code := m.Run()
+	defer os.Exit(code)
```

Comment:
Actually, as you noted, this is not needed:
As `MergeCoverages()` now returns an error and don’t `log.Fafal()`, the only part where we can "stop" the normal exit process is with your `defer os.Exit(24)`, which is garded by the `if code ==  0`, so we can safely remove this line.